### PR TITLE
docs(reference/hooks): add information about prehandler

### PR DESCRIPTION
**Pull request by RjManhas**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: docs/Reference/Hooks.md [hunk 1]
```diff
@@ -130,6 +130,9 @@ fastify.addHook('preValidation', async (request, reply) => {
 ```
 
 ### preHandler
+
+The `beforeHandler` option allows you to specify a function that is executed before the main route handler.
```
Comment (Eomm) [inline docs/Reference/Hooks.md:134]:
```suggestion
The `preHandler` hook allows you to specify a function that is executed before routes' handler.
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]
What @Eomm said.

**Comment by RjManhas** [comment]
i have fixed the suggestions.

**Review by Eomm — APPROVED** [review]

File: docs/Reference/Hooks.md [hunk 2]
```diff
@@ -130,6 +130,9 @@ fastify.addHook('preValidation', async (request, reply) => {
 ```
 
 ### preHandler
+
+The `preHandler` hook allows you to specify a function that is executed before routes' handler.
```
Comment (Uzlopak) [inline docs/Reference/Hooks.md:134]:
```suggestion
The `preHandler` hook allows you to specify a function that is executed before
routes' handler.
```

File: docs/Reference/Hooks.md [hunk 3]
```diff
@@ -130,6 +130,10 @@ fastify.addHook('preValidation', async (request, reply) => {
 ```
 
 ### preHandler
+
+The `preHandler` hook allows you to specify a function that is executed before
+routes' handler.
```
Comment (jsumners) [inline docs/Reference/Hooks.md:135]:
```suggestion
a routes's handler.
```

**Review by jsumners — APPROVED** [review]

**Review by RjManhas — COMMENTED** [review]
did the changes

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**